src/source_code/locator.rs

```diff
@@ -1,65 +1,125 @@
 //! Struct used to efficiently slice source code at (row, column) Locations.
 
-use std::borrow::Cow;
-
 use once_cell::unsync::OnceCell;
-use ropey::Rope;
 use rustpython_ast::Location;
 
 use crate::ast::types::Range;
 
 pub struct Locator<'a> {
     contents: &'a str,
-    rope: OnceCell<Rope>,
+    offsets: OnceCell<Vec<Vec<usize>>>,
+}
+
+pub fn compute_offsets(contents: &str) -> Vec<Vec<usize>> {
+    let mut offsets = Vec::new();
+    let mut current_row = Vec::new();
+    let mut current_byte_offset = 0;
+    for char in contents.chars() {
+        current_row.push(current_byte_offset);
+        if char == '\n' {
+            offsets.push(current_row);
+            current_row = Vec::new();
+        }
+        current_byte_offset += char.len_utf8();
+    }
+    offsets.push(current_row);
+    offsets
+}
```

Comment:
Use `char_indices()` might be cleaner?

```suggestion
pub fn compute_offsets(contents: &str) -> Vec<Vec<usize>> {
    let mut offsets = Vec::new();
    let mut current_row = Vec::new();
    for (index, char) in contents.char_indices() {
        current_row.push(index);
        if char == '\n' {
            offsets.push(current_row);
            current_row = Vec::new();
        }
    }
    offsets.push(current_row);
    offsets
}
```